.github/workflows/api-docs-ml.yml

```diff
@@ -0,0 +1,46 @@
+name: Generate machine learning API docs
+on:
+  pull_request_target:
+    paths:
+     - 'x-pack/plugins/ml/common/openapi/ml_apis_v3.yaml'
+    types:
+      - opened
+      - synchronize
+      - reopened
+    brances:
+      - main
+      - 8.6
+jobs:
+  ml-api-docs-preview:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          token: ${{secrets.KIBANAMACHINE_TOKEN}}
+      - name: Install libraries
+        run: |
+          npm install -g yarn
+          yarn global add @openapitools/openapi-generator-cli standard-version
```

Comment:
Is this something we can work the ML or operations team on adding to the kibana repo?  An entrypoint of `node scripts/openapi` or similar to support devs not familiar with the process.